augur/db/import_/sqlite3.py

```diff
@@ -0,0 +1,92 @@
+from augur.io.sequences import Sequences
+from augur.io.sqlite3 import Sqlite3Database
+
+
+def import_(metadata_file:str, metadata_table_name:str,
+        sequences_file:str, sequences_table_name:str,
+        db_file:str, skip_strain_indexing:bool):
+    """Import data into a SQLite3 database file."""
+    database = Sqlite3Database(db_file)
+
+    if bool(metadata_file and metadata_table_name):
+        metadata = Metadata(metadata_file)
+        import_metadata(metadata, metadata_table_name, database)
+        if not skip_strain_indexing:
+            create_index(metadata, metadata_table_name, database)
+
+    if sequences_file and sequences_table_name:
+        # Get ID name from metadata if available.
+        id_name = metadata.id_name if bool(metadata_file and metadata_table_name) else None
+
+        sequences = Sequences(sequences_file, id_name)
+        import_sequences(sequences, sequences_table_name, database)
+        if not skip_strain_indexing:
+            create_index(sequences, sequences_table_name, database)
+
+
+def import_metadata(metadata:Metadata, table_name:str, database:Sqlite3Database):
+    """Import metadata into a SQLite3 database."""
+    try:
+        database.create_table(table_name, metadata.names)
+    except sqlite3.OperationalError as e:
+        database.delete()
+        raise AugurError(f'Failed to create table: {e}')
+
+    insert_statement = f"""
+        INSERT INTO {table_name}
+        VALUES ({','.join(['?' for _ in metadata.names])})
+    """
+    rows = metadata.rows()
+    try:
+        database.executemany(insert_statement, rows)
+    except sqlite3.ProgrammingError as e:
+        database.delete()
+
+        # Translate cryptic errors into meaningful messages.
+
+        column_count_error = re.match(r"Incorrect number of bindings supplied\. The current statement uses ([0-9]+), and there are ([0-9]+) supplied\.", str(e))
+        if column_count_error:
+            expected, observed = column_count_error.groups()
+            # TODO: detect and output the bad row for easier user debugging
+            raise AugurError(f'Failed to import {metadata.file}. A row has {observed} column(s) when {expected} are expected.')
+
+        raise AugurError(f'Failed to import {metadata.file}.')
+
+
+def import_sequences(sequences:Sequences, table_name:str, database:Sqlite3Database):
+    columns = [sequences.id_name, DEFAULT_SEQUENCES_SEQ_NAME]
+    try:
+        database.create_table(table_name, columns)
+    except sqlite3.OperationalError as e:
+        database.delete()
+        raise AugurError(f'Failed to create table: {e}')
+
+    insert_statement = f"""
+        INSERT INTO {table_name}
+        VALUES ({','.join(['?' for _ in columns])})
+    """
+    records = ((record.id, str(record.seq)) for record in sequences.get_seqrecords())
+    try:
+        database.executemany(insert_statement, records)
+    except sqlite3.ProgrammingError as e:
+        print(e)
+        database.delete()
```

Comment:
This `print()` should go to stderr. Also, should probably include a stack trace. Maybe just do:

```python
database.delete()
raise e from None
```

Reply:
Fixed in 52f9590d45eb27a2808617a4590c77ed2ec25d24.